tools/reconnect_hosts.rb

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env ruby
+require File.expand_path('../config/environment', __dir__)
+require "optimist"
+
+opts = Optimist.options do
+  opt :ems_id, "The ID of the ExtManagementSystem to reconnect Hosts for", :type => :integer
+  opt :ems_name, "The name of the ExtManagementSystem to reconnect Hosts for", :type => :string
+  opt :by, "Property to treat as unique, defaults to :hostname", :type => :string, :default => "hostname"
+  opt :dry_run, "Just print out what would be done without modifying anything", :type => :boolean, :default => true
+end
+
+if opts[:ems_id].nil? && opts[:ems_name].nil?
+  Optimist.die :ems_id, "Must pass either --ems-id or --ems-name"
+end
+
+ems = if opts[:ems_id].present?
+        ExtManagementSystem.find_by(:id => opts[:ems_id])
+      else
+        ExtManagementSystem.find_by(:name => opts[:ems_id])
+      end
+
+if ems.nil?
+  print "Failed to find EMS [#{opts.key?(:ems_id) ? opts[:ems_id] : opts[:ems_name]}]"
+  exit
+end
+
+ems_id = ems.id
+
+# Find all Hosts which match the unique key
+hosts_index = Host.where(opts[:by] => ems.hosts.pluck(opts[:by])).group_by { |host| host.send(opts[:by]) }
+
+# Select where there are exactly two with the same unique property, one archived and one active
+# duplicate_hosts = hosts_index.select { |_by, hosts| hosts.count == 2 && hosts.select(&:active).count == 1 }
```

Comment:
@agrare It turns out that `Host` doesn't have an `active` method. so I sorted them in `created_on` and assigned the newer one as the active one. Not sure if there's a better way here.

Reply:
Ah yeah there is `has_active_ems?` but we should add a scope similar to vm_or_template for consistency.